server/channels/app/content_flagging_setup_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetContentFlaggingGroupID(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("should register content flagging group", func(t *testing.T) {
+		groupID, err := th.App.GetContentFlaggingGroupID()
+		require.NoError(t, err)
+		require.NotEmpty(t, groupID)
+
+		// resetting cached value of group ID to ensure second function call goes to DB layer
+		contentFlaggingGroupID = ""
+
+		// should not update the group ID when called again
+		groupID2, err := th.App.GetContentFlaggingGroupID()
+		require.NoError(t, err)
+		require.Equal(t, groupID, groupID2)
+	})
+
+	t.Run("returns cached value if it exists", func(t *testing.T) {
+		contentFlaggingGroupID = "cached-group-id"
+		groupID, err := th.App.GetContentFlaggingGroupID()
+		require.NoError(t, err)
+		require.Equal(t, "cached-group-id", groupID)
+	})
```

Comment:
I see you are trying to add some really thorough testing but I'm not sure if this is effective.

Reply:
Removed as this is not longer needed